Control over children ordering

I am wondering if any guarantees on the order of children would be a goal or non-goal of this library?

It seems `insert_with_parent` currently only pushes to the end of the children list.
If I would like to replace a child and keep the current position, this does not seem to be easily possible and I don't even know, if I can rely on `insert_with_parent` always appending at the end, because this is not documented.

For my use case I just have a left and a right child and it would be great to be able to differentiate between those. But I have no good idea, how to provide this functionality in the context of this library with an arbitrary amount of child nodes.

Possible API suggestions:
- Split insert into two functions push at the front and back of the children respectively.
    - Not a very nice API in my opinion. The use case is not very obvious at first.
    - Actual sorting would be difficult to emulate given only these functions
    - Enough for my use case
    - Easy to implement
- Allow to sorting of children via a closure
    - Makes a good API
    - Can be used to reorder children, after "replacing" (remove/insert), but needs to track old positions manually
    - That means not very nice for my use case
    - Easy to implement (https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.sort_by_key)
- Allow swapping of children
    - Also not a very nice API. The use case is not very obvious at first.
    - Actual sorting would be difficult to emulate given only these functions
    - Enough for my use case
    - Easy to implement
- Allow replacement of `Node`s and swapping of `Node`s.
    - Swapping could take two `NodeId`s
    - Replacing could insert a new node, removing the old in the process.
    - Still does not make very strong order guarantees except to keep the old order
    - Feels like it has use cases beyond this

I mostly opened this issue to discuss, if you would want to support any order guarantees anyway. The rest is just a little bit of brainstorming, please feel free to edit/extend/ignore it, however you like.

Well, originally I had this:
```
pub fn sort_children_by_data(&mut self, node_id: &NodeId) -> Result<(), NodeIdError> where T: Ord {
        let (is_valid, error) = self.is_valid_node_id(node_id);
        if !is_valid {
            return Result::Err(error.expect("Tree::move_node_to_parent: Missing an error value on finding an invalid NodeId."));
        }

        let mut children = self.get_mut_unsafe(node_id).children_mut();

        children.sort_by_key(|a| {
            self.get_unsafe(a).data()
        });

        Result::Ok(())
    }
```
and I got this error:
```
error[E0502]: cannot borrow `self` as immutable because `*self` is also borrowed as mutable
   --> src\tree.rs:455:30
    |
453 |         let mut children = self.get_mut_unsafe(node_id).children_mut();
    |                            ---- mutable borrow occurs here
454 | 
455 |         children.sort_by_key(|a| {
    |                              ^^^ immutable borrow occurs here
456 |             self.get_unsafe(a).data()
    |             ---- borrow occurs due to use of `self` in closure
...
460 |     }
    |     - mutable borrow ends here

```
So that's why I opted for the immutable borrow on the node (note `get_mut_unsafe` vs `get_unsafe`) + clone the children idea.

But maybe there's another way to approach it that I'm just not seeing right now.

--- a/src/tree.rs
+++ b/src/tree.rs
@@ -629,11 +629,14 @@ impl<T> Tree<T> {
     ///
     /// Swaps two `Node`s including their children given their `NodeId`s.
     ///
-    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    /// If one `Node` is a descendant of the other getting swapped, the former *upper* `Node` is attached as the last child of the former *lower* `Node` after the swap. (The *lower* will take the *uppers* original position as usual.)
+    /// The subtree of the former *upper* node is not touched except that the *lower* `Node` is moved including all its children.
     ///
-    /// Notes:
-    ///     - Both `NodeId`s are still valid after this process and are not swapped.
-    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    /// Both `NodeId`s are still valid after this process and are not swapped.
+    ///
+    /// This keeps the positions of the `Node`s in their parents' children collection.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
     ///
     /// ```
     /// use id_tree::Tree;

